engines/hypno/arcade.cpp

```diff
@@ -0,0 +1,342 @@
+	debugC(1, kHypnoDebugParser, "Parsing %s/%s", prefix.c_str(), filename.c_str());
+	parse_arc(data.c_str());
+	Level level;
+	level.arcade = g_parsedArc;
+	level.arcade.prefix = prefix;
+	_levels[filename] = level;
+	g_parsedArc.background.clear();
+	g_parsedArc.player.clear();
+	g_parsedArc.shoots.clear();
+}
+
+ShootSequence HypnoEngine::parseShootList(const Common::String &filename, const Common::String &data) {
+	debugC(1, kHypnoDebugParser, "Parsing %s", filename.c_str());
+	Common::StringTokenizer tok(data, " ,\t");
+	Common::String t;
+	Common::String n;
+	ShootInfo si;
+	ShootSequence seq;
+	while (!tok.empty()) {
+		t = tok.nextToken();
+		if (t[0] == '\n')
+			continue;
+		n = tok.nextToken();
+		if (t == "Z")
+			break;
+
+		Common::replace(n, "\nS", "");
+		Common::replace(n, "\nZ\n", "");
+		si.name = n;
+		si.timestamp = atoi(t.c_str());
+		seq.push_back(si);
+		debugC(1, kHypnoDebugParser, "%d -> %s", si.timestamp, si.name.c_str());
+	}
+	return seq;
+}
+
+void HypnoEngine::drawPlayer() { error("Function \"%s\" not implemented", __FUNCTION__); }
+void HypnoEngine::drawHealth() { error("Function \"%s\" not implemented", __FUNCTION__); }
+void HypnoEngine::drawShoot(const Common::Point &target) { error("Function \"%s\" not implemented", __FUNCTION__); }
+
+void HypnoEngine::hitPlayer() {
+	// if the player is hit, play the hit animation
+	if (_playerFrameIdx < _playerFrameSep)
+		_playerFrameIdx = _playerFrameSep;
+}
+
+void HypnoEngine::runArcade(ArcadeShooting &arc) {
+
+	Common::Point mousePos;
+	Common::List<uint32> shootsToRemove;
+	ShootSequence shootSequence = arc.shootSequence;
+
+	_font = FontMan.getFontByUsage(Graphics::FontManager::kConsoleFont);
+	_levelId = arc.id;
+	_shootSound = arc.shootSound;
+	_score = 0;
+	_health = arc.health;
+	_maxHealth = _health;
+	_defaultCursor = "arcade";
+	_shoots.clear();
+	_playerFrames = decodeFrames(arc.player);
+	_playerFrameSep = 0;
+
+
+	for (Frames::iterator it =_playerFrames.begin(); it != _playerFrames.end(); ++it) {
+		if ((*it)->getPixel(0, 0) == _pixelFormat.RGBToColor(0, 255, 255))
+			break; 
+		if ((*it)->getPixel(0, 0) == _pixelFormat.RGBToColor(0, 0, 255))
+			break; 
+
+		_playerFrameSep++;
+	}
+
+	if(_playerFrameSep == _playerFrames.size())
+		error("No player separator frame found!");
+	debugC(1, kHypnoDebugArcade, "Separator frame found at %d", _playerFrameSep);
+
+	_playerFrameIdx = -1;
+
+	MVideo background = MVideo(arc.background, Common::Point(0, 0), false, false, false);
+
+	changeCursor("arcade");
+	playVideo(background);
+	bool shootingPrimary = false;
+	bool shootingSecondary = false;
+
+	Common::Event event;
+	while (!shouldQuit()) {
+
+		while (g_system->getEventManager()->pollEvent(event)) {
+			mousePos = g_system->getEventManager()->getMousePos();
+			// Events
+			switch (event.type) {
+
+			case Common::EVENT_QUIT:
+			case Common::EVENT_RETURN_TO_LAUNCHER:
+				break;
+
+			case Common::EVENT_KEYDOWN:
```

Comment:
It would be awesome to make this engine use the KeyMapper right away, but it is optional

Reply:
This is in my TODO list.